playground/worker/dep-self-reference-url-worker/worker.js

```diff
@@ -0,0 +1,14 @@
+// copy of playground/worker/self-reference-url-worker.js
+self.addEventListener('message', (e) => {
+  if (e.data === 'main') {
+    const selfWorker = new Worker(new URL('./worker.js', import.meta.url), {
```

Comment:
In emscripten we use the self-references a lot (to start pthread workers).  

Would it be possible to allow the shorter form of this which is the even more explicit self reference:  `new URL(import.meta.url)`.   Its also more portable to write this shorter form since its robust against the containing file getting renamed.

Reply:
Yeah, that should work as Vite would skip analysis and don't have to rewrite it.
What Vite does is to detect recursive worker of `new URL("./some-file", import.meta.url)` and rewrite it into `new URL(self.location.href)`, so the runtime can reference it easily.

I confirmed it on my repro, but if you see something different, please submit a separate issue to track it.